src/main/java/net/minecraftforge/common/data/SoundsProvider.java

```diff
@@ -0,0 +1,164 @@
+package net.minecraftforge.common.data;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DirectoryCache;
+import net.minecraft.data.IDataProvider;
+import org.apache.commons.lang3.text.translate.JavaUnicodeEscaper;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+@SuppressWarnings("deprecation")
+public abstract class SoundsProvider implements IDataProvider {
+    private final Gson GSON = (new GsonBuilder()).setPrettyPrinting().disableHtmlEscaping().create();
+    private final Map<String, Sounds> data = new TreeMap<>();
+    private final DataGenerator gen;
+    private final String modid;
+
+    public SoundsProvider(DataGenerator gen, String modid) {
+        this.gen = gen;
+        this.modid = modid;
+    }
+
+    protected abstract void addSounds();
+
+    @Override
+    public void act(DirectoryCache cache) throws IOException {
+        addSounds();
+        if (!data.isEmpty())
+            save(cache, data, this.gen.getOutputFolder().resolve("assets/" + modid + "/sounds.json"));
+    }
+
+    @Override
+    public String getName() {
+        return "Sounds.json";
+    }
+
+    private void save(DirectoryCache cache, Object object, Path target) throws IOException {
+        String data = GSON.toJson(object);
+        System.out.println(data);
+        data = JavaUnicodeEscaper.outsideOf(0, 0x7f).translate(data); // Escape unicode after the fact so that it's not double escaped by GSON
```

Comment:
This is unnecessary.

Reply:
But what is unnecessary? the "System.out.println(data);" line ?